Components/RecentFiles.cpp

```diff
@@ -0,0 +1,97 @@
+#include "RecentFiles.h"
+
+#include <QFileInfo>
+#include <QSettings>
+#include <QString>
+
+RecentFiles::RecentFiles(QPointer<MainWindow> mainWindow, QPointer<QMenu> menuRecent,

```

Comment:
FYI, this isn't how I pictured this relationship (although I'm fine with it). What I thought you would do is put this file in charge of the recent files list (and negotiating it with the OS), but let Main own the menu and register the listeners. Instead, you're having Main create the menu and then this class registers with it. I would argue that this class should own the menu if it's making fundamental assumptions about the menu existing and how the menu will work. Just my 2¢.

Reply:
Right, I actually started rewriting it to do that but then was afraid if I put the connect code back into MainWindow you'd disagree so I wasn't sure. I'm still kind of not sure. However, it would simplify it a bit as I wouldn't have to pass as many constructor arguments.